### ISSUE TITLE
Linter shows false mistake in Ingress config

### ISSUE THREAD

**Issue by RafaelAybar** [Azure/vscode-kubernetes-tools#621]
I have the following config:
```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: daas-php-fpm
  namespace: develop
  annotations:
    kubernetes.io/ingress.class: "nginx"
spec:
  rules:
    - host: example.com
    - http:
        paths:
          - path: /
            backend:
              serviceName: php-fpm-nginx
              servicePort: 80
```
Visual Studio Code version: ```1.38.1
b37e54c98e1a74ba89e03073e5a3761284e3ffb0
x64 (snap)```

Plugin version: `Kubernetes 1.0.4`
But I see this issue with no reason, as far i know it is configured as official docs says:
![imagen](https://user-images.githubusercontent.com/21227176/65418196-3314a980-ddfc-11e9-85df-3c4b8c478788.png)

**Comment by itowlson** [comment]
This looks like the same underlying issue as #618.

Out of interest did this happen on older versions?  We are using 'live' Swagger for validation now, and I'm wondering if the Swagger we get from the cluster is somehow different from what we had when we used a static Swagger schema.

**Comment by RafaelAybar** [comment]
If i am not mistaken, in older versions i had not any issues like this

**Comment by itowlson** [comment]
@RafaelAybar I believe this is due to incorrect handling of the k8s `IntOrString` type by the Red Hat YAML extension.  I can try to patch this while we request an upstream fix, but at the moment I can't reproduce it in my dev environment.  Can you let me know what OS you're on and the version of the YAML extension you have installed, please?  I'm trying to isolate a repro scenario so I can verify a fix.  Thanks!

cc @huhamhire @shreyu86 since it looks like you hit the same issue!

**Comment by shreyu86** [comment]
@itowlson I'll add a gist that can be used to test tomorrow. Thanks for looking into this.

**Comment by shreyu86** [comment]
@itowlson here is the gist https://gist.github.com/shreyu86/356cb039bccc70bd795d415a7c538890 
```
Version: 1.39.2
Commit: 6ab598523be7a800d7f3eb4d92d7ab9a66069390
Date: 2019-10-15T15:33:00.827Z
Electron: 4.2.10
Chrome: 69.0.3497.128
Node.js: 10.11.0
V8: 6.9.427.31-electron.0
OS: Darwin x64 19.0.0
```

redhat.vscode-yaml vesion: 0.5.3
ms-kubernetes-tools.vscode-kubernetes-tools version: 1.0.4

**Comment by itowlson** [comment]
@shreyu86 Thanks!  One more question, what is the version of Kubernetes on the active cluster?

**Comment by itowlson** [comment]
@shreyu86 Also, I see you're on Mac OS; do you have any remote extensions or SSH in play, or is everything (except the k8s hosting) happening on the Mac?  What k8s are you using e.g. AKS, Minikube, Kind, GKE, etc.?  Sorry for the random questions, trying to track down possible environmental factors because this stubbornly refuses to reproduce for me!

**Comment by shreyu86** [comment]
@itowlson I have SSH extensions installed but everything in this case is happening locally on Mac OS.  Using EKS for K8s. Happy to provide more context too.

**Comment by RafaelAybar** [comment]
@itowlson i am using `Linux WS189 5.0.0-31-generic #33~18.04.1-Ubuntu SMP Tue Oct 1 10:20:39 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux`

**Comment by itowlson** [comment]
Thanks @RafaelAybar.  Could I also ask what Kubernetes version you are running on your cluster?  Thanks!

**Comment by itowlson** [comment]
Fix on its way (#636) - thanks for your patience!

**Comment by itowlson** [comment]
Closed via #636.
